godtools/Managers/LanguagesManager.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  LanguagesManager.swift
+//  godtools
+//
+//  Created by Ryan Carlson on 4/18/17.
+//  Copyright © 2017 Cru. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import Alamofire
+import PromiseKit
+import Spine
+import MagicalRecord
+
+class LanguagesManager: NSObject {
+    static let shared = LanguagesManager()
+    
+    let path = "/languages"
+    let serializer = Serializer()
+    
+    var languages = [Language]()
+    
+    override init() {
+        super.init()
+        serializer.registerResource(LanguageResource.self)
+    }
+    
+    func loadFromDisk() -> Promise<[Language]> {
+        languages = Language.mr_findAll() as! [Language]
+        
+        for language in languages {
+            language.localizedName = NSLocale.current.localizedString(forLanguageCode: language.code!)
+        }
+        
+        languages = languages.sorted { (language1, language2) -> Bool in
+            return language1.localizedName!.compare(language2.localizedName!).rawValue < 0
+        }
+        
+        return Promise(value:languages)
+    }
+    
+    func loadFromRemote() -> Promise<[Language]> {
+        UIApplication.shared.isNetworkActivityIndicatorVisible = true
+        
+        return Alamofire.request(URL(string: "\(GodToolsConstants.kApiBase)/\(path)")!).responseData().then { data -> Promise<[Language]> in
+            let jsonDocument = try! self.serializer.deserializeData(data)
+            
+            if jsonDocument.data != nil {
+                MagicalRecord.save(blockAndWait: { (context) in
```

Comment:
As I wrote before, I believe all of this logic can be re-usable by adding it in an "abstract" class. I like how the protocols and extensions are encouraged by the community of Swift 3, thats a good way to make things separated and re-usable